src/app/shared/example-viewer/example-viewer.ts

```diff
@@ -59,20 +65,42 @@ export class ExampleViewer {
   private _example: string;
 
   /** Range of lines of the source code to display in compact view. */
-  @Input() lines: readonly [number, number];
+  @Input() region: string;
 
   /** Name of file to display in compact view. */
   @Input() file: string;
 
   constructor(private readonly snackbar: MatSnackBar, private readonly copier: CopierService) {
   }
 
+  ngOnInit() {
+    if (this.file) {
+      this.fileUrl = this.generateUrl(this.file);
+    }
+  }
+
+  selectCorrectTab() {
+    const fileType = this.file.split('.')[1];
```

Comment:
you should grab the last chunk, not the second - this could be different if the file is named something like `stuff.spec.ts`

Reply:
also add a test to cover this

Reply:
I'd also consider throwing an error/logging a warning if the file type is something other than the ones listed.